eventmesh-protocol-plugin/eventmesh-protocol-meshmessage/src/main/java/org/apache/eventmesh/protocol/meshmessage/resolver/http/SendMessageRequestProtocolResolver.java

```diff
@@ -42,98 +42,64 @@ public static CloudEvent buildEvent(Header header, Body body) throws ProtocolHan
-                    .withSubject(sendMessageRequestBody.getTopic())
-                    .withType("eventmeshmessage")
-                    .withSource(URI.create("/"))
-                    .withData(content.getBytes(StandardCharsets.UTF_8))
-                    .withExtension(ProtocolKey.REQUEST_CODE, code)
-                    .withExtension(ProtocolKey.ClientInstanceKey.ENV, env)
-                    .withExtension(ProtocolKey.ClientInstanceKey.IDC, idc)
-                    .withExtension(ProtocolKey.ClientInstanceKey.IP, ip)
-                    .withExtension(ProtocolKey.ClientInstanceKey.PID, pid)
-                    .withExtension(ProtocolKey.ClientInstanceKey.SYS, sys)
-                    .withExtension(ProtocolKey.ClientInstanceKey.USERNAME, username)
-                    .withExtension(ProtocolKey.ClientInstanceKey.PASSWD, passwd)
-                    .withExtension(ProtocolKey.VERSION, version.getVersion())
-                    .withExtension(ProtocolKey.LANGUAGE, language)
-                    .withExtension(ProtocolKey.PROTOCOL_TYPE, protocolType)
-                    .withExtension(ProtocolKey.PROTOCOL_DESC, protocolDesc)
-                    .withExtension(ProtocolKey.PROTOCOL_VERSION, protocolVersion)
-                    .withExtension(SendMessageRequestBody.BIZSEQNO, sendMessageRequestBody.getBizSeqNo())
-                    .withExtension(SendMessageRequestBody.UNIQUEID, sendMessageRequestBody.getUniqueId())
-                    .withExtension(SendMessageRequestBody.PRODUCERGROUP,
-                        sendMessageRequestBody.getProducerGroup())
-                    .withExtension(SendMessageRequestBody.TTL, sendMessageRequestBody.getTtl());
-                if (StringUtils.isNotEmpty(sendMessageRequestBody.getTag())) {
-                    cloudEventBuilder = cloudEventBuilder.withExtension(SendMessageRequestBody.TAG, sendMessageRequestBody.getTag());
-                }
-                if (sendMessageRequestBody.getExtFields() != null && sendMessageRequestBody.getExtFields().size() > 0) {
-                    for (Map.Entry<String, String> entry : sendMessageRequestBody.getExtFields().entrySet()) {
-                        cloudEventBuilder = cloudEventBuilder.withExtension(entry.getKey(), entry.getValue());
-                    }
-                }
-                event = cloudEventBuilder.build();
+            switch (protocolVersion) {
+                case SpecVersion.V1.toString():
+                    cloudEventBuilder = CloudEventBuilder.v1();
+                    break;
+                case SpecVersion.V03.toString():
+                    cloudEventBuilder = CloudEventBuilder.v03();
+                    break;
+                default:
+                    return null; // unsupported protocol version
             }
+
+            CloudEvent event = getBuilderCloudEvent(sendMessageRequestHeader, sendMessageRequestBody, cloudEventBuilder);
             return event;
```

Comment:
I think this could be done here:
```
return getBuilderCloudEvent(sendMessageRequestBody, sendMessageRequestHeader, cloudEventBuilder);
```
There is no need to use a redundant intermediate amount.

Reply:
Agreed